uint32 queryIdx
uint32 trainIdx
uint32 imgIdx
float32 distance
